### PR TITLE
make server threads configurable with server.properties file

### PR DESCRIPTION
**Pull request by ahmedali6**
### Description

Fixes: https://github.com/apache/cloudstack/issues/9161

### Types of changes
- Enhancement (improves an existing feature and functionality)

#### Feature/Enhancement Scale

- Minor

### How Has This Been Tested?

Tested manually by building and running the server.

### CODE REVIEW

**Review by lucas-a-martins — APPROVED** [review]
CLGTM

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by rohityadavcloud — APPROVED** [review]
LGTM, didn't test it though

**Comment by rohityadavcloud** [comment]
Ideally max thread could be 4 or 8 times the number of CPUs, need to explore how jetty connection pooling works - in which case we may only want to configure the max threads (min can be set to 4 or 8 times number of CPUs).

File: client/src/main/java/org/apache/cloudstack/ServerDaemon.java [hunk 1]
```diff
@@ -158,8 +164,8 @@ public void init(final DaemonContext context) {
     public void start() throws Exception {
         // Thread pool
         final QueuedThreadPool threadPool = new QueuedThreadPool();
-        threadPool.setMinThreads(10);
-        threadPool.setMaxThreads(500);
+        threadPool.setMinThreads(minThreads);
+        threadPool.setMaxThreads(maxThreads);
```
Comment (sureshanaparti) [inline client/src/main/java/org/apache/cloudstack/ServerDaemon.java:168]: @ahmedali6 I think, it's better to some restriction on min & max threads, as it's configurable now (also, need to add as a comment in the properties file if possible, check if any doc update is required)

Comment (ahmedali6) [reply client/src/main/java/org/apache/cloudstack/ServerDaemon.java:168]: @sureshanaparti I added the threads config to the server. properties file. Do you have a suggestion about what  restrictions to do here?

**Pull request closed without merging**